src/entities/trades/gnosis-protocol/CoWTrade.ts

```diff
@@ -56,17 +61,20 @@ export class GnosisProtocolTrade extends Trade {
    */
   public readonly feeAmount: CurrencyAmount
 
-  constructor({
-    chainId,
-    inputAmount,
-    maximumSlippage,
-    outputAmount,
-    tradeType,
-    order,
-    fee,
-    feeAmount,
-  }: GnosisProtocolTradeConstructorParams) {
+  constructor(params: CoWTradeParams) {
+    const { chainId, feeAmount, inputAmount, maximumSlippage, outputAmount, quote, tradeType, fee } = params
+
     invariant(!currencyEquals(inputAmount.currency, outputAmount.currency), 'SAME_TOKEN')
+
+    const GPv2Settlement = contractNetworks.GPv2Settlement as Record<
+      ChainId,
+      Record<'transactionHash' | 'address', string>
+    >
+
+    const approveAddress = GPv2Settlement[chainId]?.address
+
+    invariant(approveAddress, 'Missing GPv2Settlement address')
```

Comment:
what are we doing when this function throws an error?

Reply:
We don't do anything. This is a safety check to prevent creating bad instances of `CoWTrade`.